File: py_parameter/py_parameter/roomba_node.py
import rclpy
from rclpy.node import Node 

class PublishModeOfOperation(Node):
    
    def __init__(self):
        super().__init__("roomba")
        self.declare_parameter('roomba_mode','idle')
        self.create_timer(1.0,self.status_callback)

    def status_callback(self):
        roomba_mode = self.get_parameter('roomba_mode').get_parameter_value().string_value

        if roomba_mode == 'sweep':
            self.get_logger().info("Roomba is Sweeping..")
        elif roomba_mode == 'mop':
            self.get_logger().info("Roomba is Moping...")
        elif roomba_mode == 'idle':
            self.get_logger().info("Roomba is idle...")

def main(args=None):
    rclpy.init(args=args)
    roomba = PublishModeOfOperation()
    rclpy.spin(roomba)
    roomba.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: cpp_parameters/src/roomba_node.cpp
#include <chrono>
#include <functional>
#include <string>

#include <rclcpp/rclcpp.hpp>

using namespace std::chrono_literals;

class RoombaRobot : public rclcpp::Node
{
public:
  RoombaRobot()
  : Node("roomba_node")
  {
    this->declare_parameter("roomba_mode", "idle");

    timer_ = this->create_wall_timer(
      1000ms, std::bind(&RoombaRobot::timer_callback, this));
  }

  void timer_callback()
  {
    std::string my_param = this->get_parameter("my_parameter").as_string();

    RCLCPP_INFO(this->get_logger(), "Roomba is %s!", my_param.c_str());
  }

private:
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<RoombaRobot>());
  rclcpp::shutdown();
  return 0;
}